store/plugins/preferencesSyncPlugin.ts

```diff
@@ -1,12 +1,91 @@
+import { DisplayPreferencesDto } from '@jellyfin/client-axios';
 import { Plugin } from 'vuex';
-import { AppState } from '..';
+import { AppState } from '~/store';
+import { syncedMutations } from '~/store/clientSettings';
 
 export const preferencesSync: Plugin<AppState> = (store) => {
-  store.subscribe((mutation) => {
-    if (mutation.type.includes('settings/')) {
-      store.dispatch('displayPreferencesApi/updateSettings', null, {
-        root: true
-      });
+  store.subscribe(async (mutation) => {
+    if (
+      syncedMutations.some((syncedMutation) =>
+        mutation.type.includes(syncedMutation)
+      ) &&
+      store.$auth.loggedIn
+    ) {
+      try {
+        await store.dispatch('setSyncStatus', true);
+        // We set a new last sync date at the start, so if the push fails, we still have the last attempt's
+        // date and we can compare with the server when we're back online
+        await store.dispatch('clientSettings/setLastSyncDate');
+
+        // The fetch part is done because DisplayPreferences doesn't accept partial updates
+        const response = await store.$api.displayPreferences.getDisplayPreferences(
+          {
+            displayPreferencesId: 'clientsettings',
+            userId: store.$auth.user?.Id,
+            client: 'vue'
+          }
+        );
+
+        if (response.status !== 200) {
+          throw new Error(
+            `Unexpected API response code while pushing client preferences (${response.status})`
```

Comment:
Same here, let's add info about which store was being synced. This message is also incorrect as we're getting, not pushing here. Something like this:

``fetch(storeName): Unexpected API response code while getting client preferences (${response.status})``

Reply:
We're getting the preferences in order to push them back, though, so I think that message makes more sense in this context.